Теория вероятностей и математическая статистика (семинары).  
Урок 5. Тестирование гипотез.

### Задача 1.  
Когда используется критерий Стьюдента, а когда Z –критерий?

#### Ответ:  
$Z$-критерий используется, когда $ИЗВЕСТНА$ дисперсия $\sigma$ генеральной совокупности,  
и когда размер выборки велик, т.е. n> 30:  
$Формула: Z_н=\frac{\bar X-\mu}{\sigma/\sqrt n}$    
$t$-критерий (Стьюдента) используется, когда дисперсия $\sigma$ генеральной совокупности $НЕИЗВЕСТНА$,  
и когда размер выборки мал, т.е n < 30:  
$Формула: t_н=\frac{\bar X-\mu}{\sigma_n/\sqrt n}$

### Задача 2.  
Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные
автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из
n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

### Решение:  
Сформулируем нулевую гипотезу:  
    $H0: \mu=\mu_0=17$, где $\mu$ и $\mu_0$ средние выборки и генеральной совокупности и выборки, соответственно.  
Альтернативная гипотеза:  
$H1: \mu>\mu_0$  
Поскольку нам известны и среднее и дисперсия генеральной совокупности, то для проверки гипотезы будем использовать $Z$-критерий.  
$z_н=\frac{\bar X-\mu_0}{\sigma/\sqrt n}$, где  
$\bar X=\mu=17.5$ среднее по выборке,  
$\sigma=\sqrt 4=2$ среднее квадратическое отклонение генеральной совокупности,  
$n=100$ размер выборки.

In [6]:
z=(17.5-17)/0.2
print(f'Наблюдаемое значение Z-критерия: {z: .4f}')

Наблюдаемое значение Z-критерия:  2.5000


Найдем табличное значение $Z$-критерия, для $\alpha=0.05$

In [22]:
import scipy.stats as stats
zt=stats.norm.ppf(0.95)
print(f'Табличное значение Z-критерия: {zt: .4f}')

Табличное значение Z-критерия:  1.6449


Сравним полученные значения:

In [18]:
z<zt

False

Поскольку $z_н>z_t$, **отвергаем** нулевую гипотезу на уровне значимости $\alpha=0.05$  

#### Ответ:  
**Отвергаем** нулевую гипотезу $H0: \mu_0=17$ на уровне значимости $\alpha=0.05$

### Задача 3.  
Проведите тест гипотезы.  
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек.  
Вес каждой пачки составляет:  
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.  
Известно, что их веса распределены нормально.  
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

#### Решение: 
Из условия задачи следует что нужно проводить проверку следующих гипотез:  
$H0: \mu=\mu_0=200$, где $\mu$ и $\mu_0$ средние выборки и генеральной совокупности и выборки, соответственно.  
Альтернативная гипотеза:  
$H1: \mu\neq\mu_0$  
при этом будем использовать **двусторонний** критерий c уровнем значимости $\alpha=1-0.99=0.01$  
Поскольку нам не известны ни среднее, ни дисперсия генеральной совокупности, то для проверки гипотезы будем использовать $t$-критерий.

In [24]:
import numpy as np
cookies=np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

Расчитаем наблюдаемое значение t-критерия $t_н$:

In [26]:
tn=(np.mean(cookies)-200)/(np.std(cookies,ddof=1)/len(cookies)**0.5)
print(f'Наблюдаемое значение t-критерия: {tn: .4f}')

Наблюдаемое значение t-критерия: -1.0651


Найдем табличное значение $t$-критерия $t_t$, для $\alpha/2=0.01/2=0.005$

In [27]:
tt=stats.t.ppf(0.995,len(cookies)-1)
print(f'Табличное значение t-критерия: {tt: .4f}')

Табличное значение t-критерия:  3.2498


Сравним полученные значения (поскольку у нас двусторонний критерий, то используем абсолютное значение полученного t-критерия):

In [28]:
np.abs(tn)<tt

True

Поскольку $|t_н|<t_t$, **принимаем** нулевую гипотезу на уровне значимости $\alpha=0.01$  
  
 #### Ответ:  
 **Принимаем** нулевую гипотезу $H0: \mu_0=200$ на уровне значимости $\alpha=0.01$

### Задача 4.  
Есть ли статистически значимые различия в росте дочерей?  
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165  
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163

#### Решение:  
Из условия задачи следует что нужно проводить проверку следующих гипотез:  
$H0: \mu=\mu_0$  
Альтернативная гипотеза:  
$H1: \mu\neq\mu_0$  
$\alpha=0.05$ 

In [11]:
import numpy as np
array1=np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165])
array2=np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])
a=0.05

In [27]:
import scipy.stats as stats
result=stats.ttest_rel(array1,array2)
print(f'Результат:\n{result}')

Результат:
Ttest_relResult(statistic=0.6648478531431979, pvalue=0.5228168632983574)


Поскольку p-значение теста (pvalue=0.5228) больше $\alpha=0.05$, мы не отвергаем нулевую гипотезу:  
pvalue=0.5228 > $\alpha=0.05$

#### Ответ:  
Нулевая гипотеза принимается на уровне значимости $\alpha$ так как pvalue > $\alpha$, статистически значимых различий в росте дочерей нет.